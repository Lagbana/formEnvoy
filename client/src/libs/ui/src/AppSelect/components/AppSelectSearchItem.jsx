/* Entry on settings screen
=============================== */

import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import { Typography, Colors } from 'libs/ui/src/WebButton/node_modules/libs/ui/src/WebLoading/node_modules/colibri/styles';
import { AppIcon } from '../../AppIcon';
import { AppSelectSearchItemProps } from '../AppSelect.interface';

export const AppSelectSearchItem: React.FC<AppSelectSearchItemProps> = ({
  onClose,
  onBack,
  title = '',
}) => {
  // styles

  const styles = StyleSheet.create({
    container: {
      paddingHorizontal: 10,
      paddingVertical: 5,
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
    },
    icon: {
      color: Colors.gray,
      padding: 10,
    },
    title: {
      ...Typography.bodyLarge,
      color: Colors.blue,
      paddingTop: 0,
      paddingBottom: 6,
    },
  });

  return (
    <View style={styles.container}>
      <TouchableOpacity onPress={onBack}>
        <AppIcon name="arrow-left" style={styles.icon} size={24} />
      </TouchableOpacity>
      <Text style={styles.title}>{title}</Text>
      <TouchableOpacity onPress={onBack}>
        <AppIcon name="x" style={styles.icon} size={24} />
      </TouchableOpacity>
    </View>
  );
};
