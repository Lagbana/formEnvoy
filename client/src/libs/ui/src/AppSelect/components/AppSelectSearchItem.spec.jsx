import React from 'react'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { AppSelectSearchItem } from './AppSelectSearchItem'

configure({ adapter: new Adapter() })

test('<AppSelectSearchItem /> renders correctly', () => {
  const component = shallow(
    <AppSelectSearchItem
      title={'Test title'}
      onBack={() => {}}
      onClose={() => {}}
    />
  )
  expect(toJson(component)).toMatchSnapshot()
})
