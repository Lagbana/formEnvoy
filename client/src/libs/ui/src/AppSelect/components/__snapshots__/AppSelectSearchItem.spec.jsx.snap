// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppSelectSearchItem /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingVertical": 5,
    }
  }
>
  <ForwardRef
    onPress={[Function]}
  >
    <AppIcon
      name="arrow-left"
      size={24}
      style={
        Object {
          "color": "#808898",
          "padding": 10,
        }
      }
    />
  </ForwardRef>
  <Text
    style={
      Object {
        "color": "#3485D7",
        "fontFamily": "ObjektivMk1-Regular",
        "fontSize": 20,
        "lineHeight": 28,
        "paddingBottom": 6,
        "paddingTop": 0,
      }
    }
  >
    Test title
  </Text>
  <ForwardRef
    onPress={[Function]}
  >
    <AppIcon
      name="x"
      size={24}
      style={
        Object {
          "color": "#808898",
          "padding": 10,
        }
      }
    />
  </ForwardRef>
</View>
`;
